ompi/mpi/fortran/use-mpi-f08/base/ts.c

```diff
@@ -0,0 +1,141 @@
+/* -*- Mode: C; c-basic-offset:4 ; -*- */
+/*
+ * Copyright (c) 2014      Argonne National Laboratory.
+ * Copyright (c) 2019      Research Organization for Information Science
+ *                         and Technology (RIST).  All rights reserved.
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+
+#include "ts.h"
+
+#include <assert.h>
+
+#if OMPI_FORTRAN_HAVE_TS
+
+int ompi_ts_create_datatype(CFI_cdesc_t *cdesc, int oldcount, MPI_Datatype oldtype, MPI_Datatype *newtype)
+{
+    const int MAX_RANK = 15; /* Fortran 2008 specifies a maximum rank of 15 */
+    MPI_Datatype types[MAX_RANK + 1]; /* Use a fixed size array to avoid malloc. + 1 for oldtype */
+    int mpi_errno = MPI_SUCCESS;
+    int accum_elems = 1;
+    int accum_sm = cdesc->elem_len;
+    int done = false; /* Have we created a datatype for oldcount of oldtype? */
+    int last; /* Index of the last successfully created datatype in types[] */
+    int extent;
+    int i, j;
+
+#ifdef OPAL_ENABLE_DEBUG
```

Comment:
This should be `#if` instead of `#ifdef`, right?

I see several other places where this occurs; this needs to be fixed everywhere because `OPAL_ENABLE_DEBUG` is **always** defined -- it's just always 0 or 1: `#if` vs. `#ifdef` makes a big difference.

Reply:
oh yes.  fixed that.  is ifdef OPAL_ENABLE_DEBUG in a number of other places (not touched by this PR).  I'll open a PR to fix those.

Reply:
see https://github.com/open-mpi/ompi/pull/13139